File: abc/152/5.py
def gcd(a,b):
    while b!=0:
        a,b=b,a%b
    return a

def lcm(a, b):
    return (a * b)//gcd(a,b)

N = int(input())
A = list(map(int,input().split()))

lcm_a = 1

for i in A:
    lcm_a = lcm(lcm_a,i)

ans = 0
for i in A:
    ans +=lcm_a//i 
    
mod = int(1e9+7)
print(ans%mod)

File: abc/152/5.cpp
#include<iostream>
using namespace std;

int gcd(int a, int b)
{
    if (a < b) {
        a ^= b;
        b ^= a;
        a ^= b;
    }
    
    return b ? gcd(b, a % b) : a;
}

int lcm(int a, int b)
{
    return a * b / gcd(a, b);
}

int main(void){
    long long N;
    cin>>N;
    long long A[N];
    for (int i = 0;i<N;++i){
        cin>>A[i];
    }
    long long a_lcm = 1;
    long long mod = 1e9 + 7;
    for (int i = 0;i<N;++i){
        a_lcm = lcm(a_lcm,A[i])%mod;
    }
    cout<<a_lcm;
    long long B=0;

    for (int i = 0;i<N;++i){
        B += (a_lcm/A[i]);
    }

    cout<<B%mod<<endl;
    return 0;
}